src/test/scala/nl.knaw.dans.easy.solr4files/SearchServletSpec.scala

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (C) 2017 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.solr4files
+
+import org.apache.http.HttpStatus._
+import org.apache.solr.client.solrj.response.QueryResponse
+import org.apache.solr.client.solrj.{ SolrClient, SolrRequest, SolrResponse }
+import org.apache.solr.common.params.SolrParams
+import org.apache.solr.common.util.NamedList
+import org.apache.solr.common.{ SolrDocument, SolrDocumentList }
+import org.scalamock.scalatest.MockFactory
+import org.scalatra.test.scalatest.ScalatraSuite
+
+class SearchServletSpec extends TestSupportFixture
+  with ServletFixture
+  with ScalatraSuite
+  with MockFactory {
+
+  private class StubbedWiring extends ApplicationWiring(createConfig("vault")) {
+
+    override lazy val solrClient: SolrClient = new SolrClient() {
+      // can't use mock because SolrClient has a final method
+
+      override def query(params: SolrParams): QueryResponse = mockQueryResponse
+
+      override def close(): Unit = ()
+
+      override def request(solrRequest: SolrRequest[_ <: SolrResponse], s: String): NamedList[AnyRef] =
+        throw new Exception("mocked request")
+    }
+
+    private def mockQueryResponse = {
+
+      new QueryResponse {
+        override def getResults: SolrDocumentList = new SolrDocumentList {
+          setNumFound(2)
+          setStart(0)
+          add(new SolrDocument(new java.util.HashMap[String, AnyRef] {
+            put("x", "y")
+          }))
+          add(new SolrDocument(new java.util.HashMap[String, AnyRef] {
+            put("a", "b")
+            put("c", "d") // TODO try date once we have a typeHint for the json serialization
+          }))
+        }
+      }
+    }
+  }
+
+  private val app = new EasyUpdateSolr4filesIndexApp(new StubbedWiring)
+  addServlet(new SearchServlet(app), "/*")
+
+  "get /" should "complain about missing argument" in {
+    get("/?q=something") {
+      body shouldBe "filesearch requires param 'text' (a solr dismax query), got [q -> something]"
+      status shouldBe SC_BAD_REQUEST
+    }
+  }
+
+  it should "return json" in {
```

Comment:
wouldn't this be better understandable/cleaner/clearer if you parsed your response back to JSON and then compare it to the expected JSON structure?

Reply:
The message applied to https://github.com/DANS-KNAW-jp/easy-update-solr4files-index/blob/b54b5b3a9a731bdd567d6dadf1a884ea649579d5/src/test/scala/nl.knaw.dans.easy.solr4files/SearchServletSpec.scala#L77-L92 these lines are changed, the header now has a fixed order simplifying the test.